### PR TITLE
[node] Remove block confirmations requirement in CREATE_CHANNEL

### PR DESCRIPTION
**Pull request by snario**
Implements a fix for #1172.

NOTE: We need to deploy the upgraded `ProxyFactory.sol` contract to Kovan before merging!

### CODE REVIEW

**Comment by ebryn** [comment]
@snario Should we make contract deployment part of the CI/CD pipeline?

**Comment by snario** [comment]
@ebryn Yeah, let's add an issue for that. It's not super straightforward and no tool is great for it. Truffle Teams is working on it.

**Comment by snario** [comment]
The tests in the past ran `CREATE_CHANNEL` twice which would create a new multisig address each time. However now, the same multisig address is used and so the second create channel request fails  on this line:

```ts
if (context.stateChannelsMap.has(multisigAddress)) {
  throw Error(`Found an already-setup channel at ${multisigAddress}`);
}
```

Should we change the tests or introduce a nonce per new channel creation call? @IIIIllllIIIIllllIIIIllllIIIIllllIIIIll @emansipater @cf19drofxots

**Comment by emansipater** [comment]
imo the nonce should be derived deterministically the way we have discussed for the longterm plan. For tests it should come down to what is being tested.  If it's the same parties and channel it should just re-derive the same address. If it needs to be a new state deposit address for test purposes it should follow the deterministic scheme to make the new one.

**Comment by snario** [comment]
@emansipater how does the deterministic scheme work again?

File: packages/node/src/methods/state-channel/create/controller.ts [hunk 1]
```diff
@@ -44,23 +51,44 @@ export default class CreateChannelController extends NodeController {
...
+              [`0x${Proxy.bytecode}`, networkContext.MinimumViableMultisig]
+            )
+          )
+        ]
+      ).slice(-40)
+    );
 
-    tx.wait(blocksNeededForConfirmation).then(receipt =>
-      this.handleDeployedMultisigOnChain(receipt, requestHandler, params)
+    const tx = await this.sendMultisigDeployTx(
```
[the remark this answers is not in the document]
Comment (snario) [reply packages/node/src/methods/state-channel/create/controller.ts]: There is an issue for it: https://github.com/counterfactual/monorepo/issues/1183

**Comment by cf19drofxots** [comment]
The tests pertaining to creating a channel have been updated to pass.

Now the failure is during a secure withdrawal.

```
 FAIL  test/integration/secure-withdraw.spec.ts
  Node method follows spec - withdraw
    ✕ has the right balance for both parties after withdrawal (825ms)

  ● Node method follows spec - withdraw › has the right balance for both parties after withdrawal

    expect(received).toEqual(expected)

    Expected: 1
    Received: 3

      88 |         await nodeA.call(depositReq.type, depositReq);
      89 |
    > 90 |         expect((await provider.getBalance(multisigAddress)).toNumber()).toEqual(
         |                                                                         ^
      91 |           1
      92 |         );
      93 |

      at EventEmitter.nodeA.on (test/integration/secure-withdraw.spec.ts:90:73)
```

**Comment by snario** [comment]
Fixed test R4R @cf19drofxots

**Pull request merged**